src/clientAuth/clientSecretBasic.js

```diff
@@ -6,69 +12,65 @@ class ClientSecretBasic extends ClientAuthnMethod {
   }
 
   /**
-   * @param {*} cis Request class instance
-   * @param {*} requestArgs Request arguments
-   * @param {*} httpArgs HTTP arguments
+   * @param {?ResourceRequest} cis Request class instance
+   * @param {?ClientInfo} cliInfo Client information
+   * @param {?Object<string, string>} requestArgs Request arguments
+   * @param {?Object<string, string>} httpArgs HTTP header arguments
+   * @param {?Object<string, string>} kwargs Other extra arguments
+   * @return {!Object<string, string>} HTTP header arguments
    */
-  construct(cis, cliInfo, requestArgs, httpArgs, kwargs) {
-    cliInfo = cliInfo || null;
-    requestArgs = requestArgs || null;
-    httpArgs = httpArgs || null;
-
-    if (httpArgs == null) {
-      httpArgs = {};
-    }
+  construct(cis, cliInfo = null, requestArgs = null, httpArgs = {}, kwargs) {
```

Comment:
We'll have to discuss the method signature at some point. The logic to extract the relevant properties (password, client_secret, user, etc.) seems overly complex. I'd like to see some examples of how such API is used to understand why it's this way. Could you add a global readme with such examples? Thanks!

Reply:
The logic is based off of the python library so I would suggest discussing with Roland as well. Will add a README with examples to my TODO list. For now please refer to http://oiccli.readthedocs.io/en/latest/intro.html .